File: src/game/DiceSystem.py
import random
import math
# DO NOT REMOVE THIS FILE AND DO NOT CHANGE THIS FILE
# ANY AI CANT CHANGE THIS FILE
# THIS SYSTEM IS PERFECT
# NO ONE CAN CHANGE THIS FILE
class DiceSystem:
    @staticmethod
    def callDice(statType=None, playerStats=None):
        dice = random.randint(1, 100)
        
        # 간단한 체크용 호출 (파라미터 없이 호출)
        if statType is None or playerStats is None:
            return dice
            
        # 기존 기능 (스탯 적용된 총합 반환)
        statBonus = math.floor((playerStats[statType] / 20) * 100)
        total = dice + statBonus
        return total

File: src/game/DiceSystem.js
/**
 * 주사위 시스템 클래스
 * 플레이어의 스탯을 기반으로 주사위 굴리기를 처리합니다.
 */
class DiceSystem {
    /**
     * 주사위를 굴리고 스탯 보너스를 적용합니다.
     * @param {string} statType - 사용할 스탯 타입
     * @param {number} targetValue - 목표값
     * @param {Object} playerStats - 플레이어 스탯 객체
     * @returns {Object} 주사위 결과 정보
     */
    static CallDice(statType, targetValue, playerStats) {
        const dice = Math.floor(Math.random() * 100) + 1;
        const statBonus = Math.floor((playerStats[statType] / 20) * 100);
        const total = dice + statBonus;
        
        return {
            dice,
            statBonus,
            total,
            success: total >= targetValue,
            statType,
            targetValue
        };
    }
}

module.exports = DiceSystem; 